nix/package.nix

```diff
@@ -1,31 +1,26 @@
 {
   lib,
+  stdenv,
+  cargo,
   dbus,
+  meson,
+  ninja,
   python3,
   pkg-config,
+  rustc,
   rustPlatform,
   xorg,
   rev ? "dirty",
 }: let
   cargoToml = builtins.fromTOML (builtins.readFile ../Cargo.toml);
 in
-  rustPlatform.buildRustPackage {
+  stdenv.mkDerivation {
     pname = "tailray";
     version = "${cargoToml.package.version}-${rev}";
 
-    src = lib.fileset.toSource {
-      root = ../.;
-      fileset =
-        lib.fileset.intersection
-        (lib.fileset.fromSource (lib.sources.cleanSource ../.))
-        (lib.fileset.unions [
-          ../src
-          ../Cargo.toml
-          ../Cargo.lock
-        ]);
-    };
+    src = ../.;
```

Comment:
why get rid of toSource? this makes the source non-reproducible and puts churn in the store path if I understand source paths correctly

Reply:
If I leave it there nothing is copied to the build dir.

Reply:
okay I'll change this manually then. until then could you use `builtins.path` ?